src/ucp/wireup/wireup.c

```diff
@@ -524,12 +526,12 @@ ucp_wireup_process_request(ucp_worker_h worker, ucp_ep_h ep,
             ep->conn_sn = msg->conn_sn;
             ucp_ep_match_insert(worker, ep, remote_uuid, ep->conn_sn,
                                 UCS_CONN_MATCH_QUEUE_UNEXP);
+            ucp_ep_update_remote_id(ep, msg->src_ep_id);
         } else {
+            ucp_ep_update_remote_id(ep, msg->src_ep_id);
```

Comment:
why needed?

Reply:
now, returned back - but changed other places
we rely on the fact that if REMOTE_ID is set then FLUSH state is valid
so, changed it to check FLUSH_STATE/ON_MACTCH_CTX flags when updating remote ID